### ISSUE TITLE
fix: pca transform

### ISSUE THREAD

**Issue by AFAgarap** [sentiteam/meltwater#1]
I saw in your [notebook implementation](https://github.com/sentiteam/meltwater/blob/master/notebooks/Language%20Classifier%20using%20unigram%20and%20bigram%20(PCA).ipynb) the PCA transformation of features. In the cell for "Normalizing the features and transforming it",

```
scaler = StandardScaler().fit(X)
X =  scaler.transform(X)
pca = PCA(n_components=4000)
X = pca.fit_transform(X)
```

We should have the dataset features partitioned to training and testing first. Then, fit the PCA to the training dataset. Afterwards, transform both training and testing dataset using the fitted PCA. This paradigm prevents the model from "cheating". That is, the PCA must only have information on the training dataset, not the testing dataset as well.
